app/uk/gov/hmrc/fileupload/controllers/EnvelopeConstraints.scala

```diff
@@ -16,53 +16,49 @@
 
 package uk.gov.hmrc.fileupload.controllers
 
-import uk.gov.hmrc.fileupload.read.envelope.Envelope.ContentTypes
-import uk.gov.hmrc.fileupload.read.envelope._
+import uk.gov.hmrc.fileupload.write.envelope.EnvelopeHandler.ContentTypes
 
-object EnvelopeConstraints {
+case class EnvelopeConstraints (maxItems: Int,
+                                maxSize: Size,
+                                maxSizePerItem: Size,
+                                contentTypes: List[ContentTypes]) {
 
-  private val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
+  val maxSizeInBytes: Long = maxSize.inBytes
+  val maxSizePerItemInBytes: Long = maxSizePerItem.inBytes
+}
 
-  def isAValidSize(size: String): Boolean = {
-    if (size.isEmpty) false
-    else {
-      size.toUpperCase match {
-        case sizeRegex(num, unit) =>
-          unit match {
-            case "KB" => true
-            case "MB" => true
-            case _ => false
-          }
-        case _ => false
-      }
+sealed trait SizeUnit
+case object KB extends SizeUnit
+case object MB extends SizeUnit
+
+case class Size(value: Long, unit: SizeUnit) {
+  override def toString: String = value + unit.toString
+
+  def inBytes: Long = {
+    unit match {
+      case KB => value * 1024
+      case MB => value * 1024 * 1024
     }
   }
+}
+
+object Size {
 
-  def translateToByteSize(size: String): Long = {
-    if (!isAValidSize(size)) throw new IllegalArgumentException(s"Invalid constraint input")
+  val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
+
+  def apply(asString: String) = {
+    if (asString.isEmpty) throw new IllegalArgumentException(s"Invalid constraint input")
     else {
-      size.toUpperCase match {
+      asString.toUpperCase match {
         case sizeRegex(num, unit) =>
           unit match {
-            case "KB" => num.toInt * 1024
-            case "MB" => num.toInt * 1024 * 1024
+            case "KB" => new Size(num.toInt, KB)
+            case "MB" => new Size(num.toInt, MB)
+            case _ => throw new IllegalArgumentException(s"Invalid constraint input")
           }
+        case _ => throw new IllegalArgumentException(s"Invalid constraint input")
```

Comment:
throwing exceptions for input validation in Scala is often considered an anti-pattern in Scala as it produces an impure function, it would also be good to mention _why_ the input is invalid under each case (to make it easier for the client to debug).

you could consider using something like this to create a pure validation function that makes it clear why validation failed:

```
sealed trait SizeValidationFailure {
  def message: String
}

case object EmptyInput extends SizeValidationFailure {
  override def message: String = "input was empty"
}

case class InvalidUnit(unitProvided: String) extends SizeValidationFailure {
  override def message: String = s"unit '$unitProvided' was unknown: 'KB' and 'MB' are supported"
}

case class InvalidFormat(input: String) extends SizeValidationFailure {
  override def message: String = s"input '$input' did not match supported size format of '${Size.sizeRegex}'"
}

object Size {

  val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r

  def apply(asString: String): Either[SizeValidationFailure, Size]  = {
    if (asString.isEmpty)
      Left(EmptyInput)
    else {
      asString.toUpperCase match {
        case sizeRegex(num, unit) =>
          unit match {
            case "KB" => Right(Size(num.toInt, KB))
            case "MB" => Right(new Size(num.toInt, MB))
            case _ => Left(InvalidUnit(unit))
          }
        case _ => Left(InvalidFormat(asString))
      }
    }
  }

}
```

the calling site can then be altered to process the value produced rather than potentially handling an exception

Reply:
thanks